tests/Jackalope/Test/Fixture/DBUnitFixtureXML.php

```diff
@@ -192,6 +192,9 @@ public function addNode($workspaceName, DOMElement $node)
             $name       = '';
             $namespace  = '';
         }
+        if (isset($this->namespaces[$namespace])) {
```

Comment:
for better readability, could we rename the above $namespace to $namespacePrefix and then change this to `$namespace = isset($this->namespaces[$namespace]) ? $this->namespaces[$namespacePrefix] : $namespacePrefix;` (the `??` operator is not available in all PHP versions we currently support, unfortunately).

Reply:
sure, i'll get that in right away.

The tests I'll see when and if I can make some time for it, but there is another PR i'm gona make for a similar bug as the incorrect nested joins where clause

Reply:
@dbu not sure if you're code suggestion is ok (apart from the type in the isset $namespace instead of $namespacePrefix):
`$namespace = isset($this->namespaces[$namespace]) ? $this->namespaces[$namespacePrefix] : $namespacePrefix;`

I don't think you can fallback here on $namespacePrefix, technically you should even throw an exception if $namespacePrefix is set and not found in namespaces.

Reply:
ups, yeah i mixed up things. but you are right - as this is in tests we don't have to worry about BC breaks, lets do

```
if (!isset($this->namespaces[$namespacePrefix])) {
    throw new \Exception('Undefined namespace prefix '.$namespacePrefix);
}
and then directly use `$this->namespaces[$namespacePrefix]` in the code below?
```

Reply:
ah i see you pushed. forgot about the special case... i like what you have now, seems good to me.

do things in the jcr namespace not have a namespace url?

Reply:
I'm not sure if they do, not to familiar with the spec, I do know that in the tests they use jcr:one and that is expected to be namespaced, but the root node seems exempt? Not sure why.

Reply:
and i don't remember :-) probably just historic reasons from the beginning. i don't want to mess with root though, so its fine now - thanks again!